### PR TITLE
Refresh and complete MSTest documentation

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md [hunk 1]
```diff
@@ -15,66 +15,211 @@ author: mikejo5000
...
+    {
+    }
+}
+```
+
+#### Class
+
+[ClassInitialize](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.ClassInitializeAttribute>) is called right before your class is loaded (but after static constructor) and [ClassCleanup](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.ClassCleanupAttribute>) is called right after your class is unloaded.
+
+It is possible to control the inheritance behavior: only for current class using `InheritanceBehavior.None` or for all derived classes using `InheritanceBehavior.BeforeEachDerivedClass`.
```
Comment (Mikejo5000) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:142]:
```suggestion
It is possible to control the inheritance behavior only for the current class using `InheritanceBehavior.None` or for all derived classes using `InheritanceBehavior.BeforeEachDerivedClass`.
```

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md [hunk 2]
```diff
@@ -15,66 +15,211 @@ author: mikejo5000
...
+    {
+    }
+}
+```
+
+#### Test
+
+[TestInitialize](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestInitializeAttribute>) is called right before your test is started and [TestCleanup](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestCleanupAttribute>) is called right after your test is finished.
+
+The `TestInitialize` is similar to the class constructor but is usually better fitted for long or async initializations. Note that the `TestInitialize` is always called after the constructor and called for each test (including each data row of data-driven tests).
```
Comment (Mikejo5000) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:184]:
```suggestion
The `TestInitialize` is similar to the class constructor but is usually more suitable for long or async initializations. Note that the `TestInitialize` is always called after the constructor and called for each test (including each data row of data-driven tests).
```

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md [hunk 3]
```diff
@@ -15,66 +15,211 @@ author: mikejo5000
...
+
+The `TestCleanup` is similar to the class `Dispose` (or `DisposeAsync`) but is usually better fitted for long or async cleanups. Note that the `TestCleanup` is always called just before the `DisposeAsync`/`Dispose` and called for each test (including each data row of data-driven tests).
```
Comment (Mikejo5000) [inline docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:186]:
```suggestion
The `TestCleanup` is similar to the class `Dispose` (or `DisposeAsync`) but is usually more suitable for long or async cleanups. Note that the `TestCleanup` is always called just before the `DisposeAsync`/`Dispose` and called for each test (including each data row of data-driven tests).
```

**Review by Mikejo5000 — APPROVED** [review]
Looks good. Left a couple of minor suggestions

**Comment by jborsecnik** [comment]
@Mikejo5000, I noticed that three instances similar to this don't result in a live link in GitHub preview. Should they? They just look like "TestInitialize" in the preview.

`[TestInitialize](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestInitializeAttribute>)`

Also note (related, as regards moving this to the private repo):  https://github.com/MicrosoftDocs/visualstudio-docs/pull/8570#issuecomment-1281497848

**Comment by Mikejo5000** [comment]
@jborsecnik I don't see anything wrong in the formatting, so it probably is a preview issue. @Evangelink Can you move this to private repo, see comments above? We ask that of the regular MS contributors, so we get better staging, validation, and acrolinx. You need to join MicrosoftDocs GitHub organization to access the private repo (visualstudio-docs-pr). The PR Merger team can move it if that is helpful.

File: docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md (hunk 1 above)
Comment (Evangelink) [reply docs/test/using-microsoft-visualstudio-testtools-unittesting-members-in-unit-tests.md:142]: I am not native English speaker but isn't it changing the meaning of the sentence? I want to say use `None` to have no inheritance and `BeforeEachDerivedClass` to have inheritance.

**Comment by Evangelink** [comment]
I moved the PR to the private repo. I will let you close this PR if there is nothing else to do here.

**Pull request closed without merging**